flink-clients/src/main/java/org/apache/flink/client/cli/CliFrontend.java

```diff
@@ -436,15 +440,19 @@ protected void list(String[] args) throws Exception {
 
 		final List<JobStatusMessage> runningJobs = new ArrayList<>();
 		final List<JobStatusMessage> scheduledJobs = new ArrayList<>();
+		final List<JobStatusMessage> remainingJobs = new ArrayList<>();
 		jobDetails.forEach(details -> {
 			if (details.getJobState() == JobStatus.CREATED) {
 				scheduledJobs.add(details);
-			} else {
+			} else if (details.getJobState() == JobStatus.RUNNING
```

Comment:
use `JobStatus#isGloballyTerminalState` instead. This covers all cases where a job can still be considered running.